py3status/modules/hueshift.py

```diff
@@ -0,0 +1,191 @@
+        if not self.command:
+            raise Exception(STRING_NOT_AVAILABLE)
+
+        self.hue = {
+            'blueshift': {
+                'set': lambda value: ['blueshift', '-p', '-t', str(value)],
+                'reset': lambda value: ['blueshift', '-p', '-r'],
+            },
+            'redshift': {
+                'set': lambda value: ['redshift', '-r', '-O', str(value)],
+                'reset': lambda value: ['redshift', '-r', '-x'],
+            },
+            'sct': {
+                'set': lambda value: ['sct', str(value)],
+                'reset': lambda value: ['sct']
+            },
+        }
+
+        self.minimum = max(self.minimum, 1000)
+        if self.maximum > 25000:
+            self.maximum = 25000
+        if self.command == 'sct' and self.maximum > 10000:
+            self.maximum = 10000
+        if not(self.minimum <= self.initial <= self.maximum):
+            self.initial = 6500
+
+        self.temperature = self.py3.storage_get(
+            'temperature') or self.initial
+        self.last_temperature = self.temperature
+        self.last_command = None
+        self._set_color_setter_boolean()
+        if not self.is_color_setter_running:
+            self._command('set', self.temperature)
+
+    def _command(self, action=None, value=None):
+        command = self.hue[self.command][action](value)
+        if command != self.last_command:
+            self.py3.command_run(command)
+            self.last_command = command
+            self.last_temperature = value
+        else:
+            self.py3.prevent_refresh()
+
+    def _set_color_setter_boolean(self):
+        try:
+            self.py3.command_output(self.pgrep_command)
+            self.is_color_setter_running = True
+        except:
+            self.is_color_setter_running = False
+
+    def hueshift(self):
+        self.py3.threshold_get_color(self.temperature, 'hue')
+        return {
+            'cached_until': self.py3.CACHE_FOREVER,
+            'full_text': self.py3.safe_format(
+                self.format, {
+                    'temperature': self.temperature,
+                    'command': self.command,
+                    'disabled': self.is_color_setter_running,
+                })}
+
+    def kill(self):
+        self.py3.storage_set('temperature', self.last_temperature)
+
+    def on_click(self, event):
+        self._set_color_setter_boolean()
+        if self.is_color_setter_running:
+            return
+        value = self.temperature
+        button = event['button']
+        if button == self.button_reset:
+            self._command('reset')
+            self.temperature = self.initial
+        elif button == self.button_up:
+            value += self.delta
+            if value > self.maximum:
+                value = self.maximum
+        elif button == self.button_down:
+            value -= self.delta
+            if value < self.minimum:
+                value = self.minimum
```

Comment:
imho you should have the value checking done in you set function not here.  that way any time you do a set it will be checked.  It also helps readability.
```
elif button == self.button_up:
    self._command('set', value + self.data)
elif button == self.button_down:
    self._command('set', value - self.data)
```

you'd need to change `_command()` a little but it'd make the code much more robust